function-context-this/exercise-1.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @param {string} firstName
+ * @param {string} lastName
+ */
+function NamedOne(firstName, lastName) {
+  this.firstName = firstName;
+  this.lastName = lastName;
+}
+
+const fullNameRegex = /^\w+ \w+$/g;
+/**
+ * @param {string} fullName
+ */
+function validateFullName(fullName) {
+  if (typeof fullName !== 'string') return;
+
+  return fullNameRegex.test(fullName.trim()) && fullName;
+}
```

Comment:
To my personal understanding of task, we need to use "smart" validator.
```
// There are what we have in task ---> output should be "smart", like "no: lastName missing", "no: firstName missing" or no : no space between first & last names 
namedOne.fullName = "Tom" // -> no : lastName missing 
namedOne.fullName = "TomDonnovan" // -> no : no space between first & last names 
```

Reply:
Overall good looks just perfect! Actually, I was about to ask you to use class syntax instead, but then I realized this is gonna be your next learning topic :)